.github/workflows/main.yml

```diff
@@ -86,16 +97,19 @@ env:
   test_vscode_extension: ${{ contains(github.event.pull_request.labels.*.name, 'VSCode Extension') || contains(github.event.pull_request.labels.*.name, 'ci:all-tests') || github.event_name != 'pull_request' }}
 
   # 'Ant', 'Gradle', 'Maven' and 'MX' labels trigger build-tools job
-  # see job condition (env vars don't work for job conditions https://github.com/actions/runner/issues/1189 )
+  test_build_tools: ${{ contains(github.event.pull_request.labels.*.name, 'Ant') || contains(github.event.pull_request.labels.*.name, 'Gradle') || contains(github.event.pull_request.labels.*.name, 'Maven') || contains(github.event.pull_request.labels.*.name, 'MX') || contains(github.event.pull_request.labels.*.name, 'ci:all-tests') || github.event_name != 'pull_request' }}
 
-  # special commands:
-  # 'ci:all-tests' enables everything
-  # 'ci:no-build' disables the build job (and test jobs too)
-  # 'ci:dev-build' produces an artifact containing a runnable NetBeans zip distribution
-  
-  # default java distribution used by the setup-java action
-  # see https://github.com/actions/setup-java#supported-distributions
-  default_java_distribution: 'zulu'
+  # 'git', 'subversion' and 'mercurial' labels trigger versioning job
+  test_versioning: ${{ contains(github.event.pull_request.labels.*.name, 'git') || contains(github.event.pull_request.labels.*.name, 'subversion') || contains(github.event.pull_request.labels.*.name, 'mercurial') || contains(github.event.pull_request.labels.*.name, 'ci:all-tests') || github.event_name != 'pull_request' }}
+
+  # 'Java EE/Jakarta EE', 'Micronaut' and 'enterprise' labels trigger enterprise job
+  test_enterprise: ${{ contains(github.event.pull_request.labels.*.name, 'Java EE/Jakarta EE') || contains(github.event.pull_request.labels.*.name, 'Micronaut') || contains(github.event.pull_request.labels.*.name, 'enterprise') || contains(github.event.pull_request.labels.*.name, 'ci:all-tests') || github.event_name != 'pull_request' }}
+
+  # 'JavaScript', 'TypeScript', 'HTML', 'CSS', 'CSL' and 'web' labels trigger enterprise job
+  test_web: ${{ contains(github.event.pull_request.labels.*.name, 'JavaScript') || contains(github.event.pull_request.labels.*.name, 'TypeScript') || contains(github.event.pull_request.labels.*.name, 'HTML') || contains(github.event.pull_request.labels.*.name, 'CSS') || contains(github.event.pull_request.labels.*.name, 'CSL') || contains(github.event.pull_request.labels.*.name, 'web') || contains(github.event.pull_request.labels.*.name, 'ci:all-tests') || github.event_name != 'pull_request' }}
+
+  # 'tests' label activates an extra step which builds all tests
+  test_tests: ${{ contains(github.event.pull_request.labels.*.name, 'tests') || contains(github.event.pull_request.labels.*.name, 'ci:all-tests') || github.event_name != 'pull_request' }}
```

Comment:
@sdedic since I believe you asked me about this before: this would change the enterprise job to be also opt-in. It listens to https://github.com/apache/netbeans/labels/enterprise, https://github.com/apache/netbeans/labels/Java%20EE%2FJakarta%20EE and https://github.com/apache/netbeans/labels/Micronaut after this change.

analog to the build-tools, web and versioning jobs